tools/integration-test-baseline.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+require_once __DIR__ . '/../vendor/autoload.php';
+
+copyTests(__DIR__ . '/../tests');
+
+function copyTests(string $directory): void
+{
+    $finder = new \Symfony\Component\Finder\Finder();
+    $finder
+        ->directories()
+        ->in($directory)
+        ->depth('== 0');
+
+    $tests = [];
+
+    foreach ($finder as $dir) {
+        if (!file_exists($dir->getPathname() . '/input')) {
+            copyTests($dir->getPathname());
+            continue;
+        }
+        $tempDir = $dir->getPathname() . '/temp';
+        $expectedDir = $dir->getPathname() . '/expected';
+        if (!file_exists($tempDir)) {
```

Comment:
Should probably be "is_dir" instead to make the check cleaner?

I wonder though - how exactly does this work? Where does "temp" come from? It's not created from this script?!

Reply:
the temp is created from running the integration tests

Reply:
So the workflow to use this script would be:

* run all tests
* run `php tools/integration-test-baseline.php`?

Maybe then the error message could be updated to say "No temp directory exists, please run all tests (make pre-commit-tests) before executing this script" or something like that?

Reply:
I would just make the make command so that it first runs test-integration and then my script?